demos/share-files.html

```diff
@@ -0,0 +1,144 @@
+        <td><input id="text" value="The message" size="40"/></td>
+    </tr>
+    <tr><td>URL:</td>
+        <td><input type="checkbox" id="url_checkbox" checked/></td>
+        <td><input id="url" value="https://example.com" size="40"/></td>
+    </tr>
+    <tr><td>Files:</td>
+        <td><!--Without column 2, the whole table is small on mobile.--></td>
+        <td><input id="files" type="file" multiple></td>
+    </tr>
+  </table>
+  <p><input id="share" type="button" value="Share" />
+     <input id="share-no-gesture" type="button" value="Share without user gesture" /></p>
+  <div id="output"></div>
+  <p>This is a test page for the <a href="https://github.com/WICG/web-share">Web
+  Share API</a>. It will only work in browsers that have implemented the draft
+  proposal. At the time of writing, this works with:</p>
+  <ul>
+    <li>Google Chrome for Android 61 or higher (older versions can enable
+    chrome://flags#enable-experimental-web-platform-features).</li>
+    <li>Google Chrome for Chrome OS, Windows, or Linux, with
+    chrome://flags#enable-experimental-web-platform-features enabled, in
+    conjunction with a supported
+    <a href="https://github.com/WICG/web-share-target">web share target</a>
+    (<a href="https://wicg.github.io/web-share-target/demos/sharetarget.html">demo app</a>).</li>
+  </ul>
+  <script>
+    'use strict';
+
+    function sleep(delay) {
+      return new Promise(resolve => {
+        setTimeout(resolve, delay);
+      });
+    }
+
+    function logText(message, isError) {
+      if (isError)
+        console.error(message);
+      else
+        console.log(message);
+
+      const p = document.createElement('p');
+      if (isError)
+        p.setAttribute('class', 'error');
+      document.querySelector('#output').appendChild(p);
+      p.appendChild(document.createTextNode(message));
+    }
+
+    function logError(message) {
+      logText(message, true);
+    }
+
+    function checkboxChanged(e) {
+      const checkbox = e.target;
+      const textfield = document.querySelector('#' + checkbox.id.split('_')[0]);
+
+      textfield.disabled = !checkbox.checked;
+      if (!checkbox.checked)
+        textfield.value = '';
+    }
+
+    async function testWebShare() {
+      if (navigator.share === undefined) {
+        logError('Error: Unsupported feature: navigator.share()');
+        return;
+      }
+
+      const title_input = document.querySelector('#title');
+      const text_input = document.querySelector('#text');
+      const url_input = document.querySelector('#url');
+      const file_input = document.querySelector('#files');
+
+      const title = title_input.disabled ? undefined : title_input.value;
+      const text = text_input.disabled ? undefined : text_input.value;
+      const url = url_input.disabled ? undefined : url_input.value;
+      const files = file_input.disabled ? undefined : file_input.files;
+      if (!navigator.canShare || !navigator.canShare({files})) {
```

Comment:
The error message needlessly occurs if no files are selected.

Enclose the `if` with
`if (files && files.length > 0) {  }`

Reply:
Done.